docs/rfc-001.md

```diff
@@ -0,0 +1,124 @@
+# Add government department brand colours to gem for use by components
+
+## Problem and background
+
+Organisation pages ([example](https://www.gov.uk/government/organisations/attorney-generals-office)) have their own brand, which includes specific colours applied to links and borders. These colours are pulled in from [govuk_frontend_toolkit](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/stylesheets/colours/_organisation.scss) using a [mixin](https://github.com/alphagov/whitehall/blob/master/app/assets/stylesheets/frontend/views/_organisations.scss#L855-L858) to generate a range of classes.
+
+Most organisations have one colour for borders and one for links, even if they are broadly the same colour. These are applied to the existing structure of the frontend of whitehall.
+
+The structure of the new organisation pages (to be built in collections) uses components, which are individual and isolated. To reproduce this approach in the same way would mean adding all of the CSS for all of the colour options to all of the components, dramatically increasing the size of the CSS involved and introducing much duplication.
+
+## Proposed solution
+
+### In the gem
+
+All colour classes are included in a helper Sass file in the gem.
+
+Sass in the gem looks like this:
+
+```
+@import "colours/organisation";
+
+@mixin organisation-brand-colour($selector, $property, $websafe: false) {
+  @each $organisation in $all-organisation-brand-colours {
+    .color-#{$selector}--#{nth($organisation, 1)} .color-#{$selector} {
+      @if $websafe {
+        #{$property}: nth($organisation, 3);
+      }
+      @else {
+        #{$property}: nth($organisation, 2);
+      }
+    }
+  }
+}
+
+@include organisation-brand-colour("link", color, true);
+@include organisation-brand-colour("border", border-color);
+```
+
+This is included in the `_all_components.scss` file.
+
+CSS in the gem looks like this:
+
+```
+.color-link--department-for-work-and-pensions .color-link {
+  border-color: #8f1111;
+}
+
+.color-border--department-for-work-and-pensions .color-border {
+  border-color: #9f1888;
+}
+```
+
+etc.
+
+### In the component
+
+Pass component parameters for link_colour and border_color as required, as strings that match the pre-defined list e.g. 'department-for-work-and-pensions'.
```

Comment:
Do we need to be specific about the intention of the passed in variables? 

So instead of:

```rb
<%= render 'govuk_publishing_components/components/title', link_colour: 'department-for-education', border_color: 'department-for-education' %>
```

Do this:

```rb
<%= render 'govuk_publishing_components/components/title', brand: 'department-for-education' %>

Reply:
Yeah, that's simpler.